src/devices/SenseHat/SenseHatLedMatrix.cs

```diff
@@ -32,6 +34,51 @@ public abstract class SenseHatLedMatrix : IDisposable
         /// </summary>
         protected const int NumberOfPixelsPerColumn = 8;
 
+        /// <summary>
+        /// Lazily intialized pixel font reference. Initialized when rendering text.
+        /// </summary>
+        protected SenseHatTextFont? _pixelFont = null;
```

Comment:
I do like the fact that we're extending this to support text scrolling etc but I'd prefer this logic was reusable to any LED matrix. Here is how I'd imagine this would work:
- this class remains mostly untouched (unless you need to add extra API or whatever to make text work then those changes are ok but nothing text related)
- text scrolling / setting is an abstraction which takes abstract display (or couple of Action/Func for each needed operation)

the usage would be more or less like this (you might have better ideas but this is bare minimum to isolate this utility):

```csharp
SenseHatLedMatrix matrix = ...;
LedMatrixTextScroller textScroller = new(width, height, (x, y, color) => matrix.DrawPixel(x, y, color));
textScroller.Start();
textScroller.SetText("hello");
textScroller.Stop();
```

Note: on first version I don't expect you to integrate anywhere else but if anyone else asks or tries to add it I'll point them in this direction so they can re-use this code

Reply:
The callback could also draw entire buffer or whatever, pick a design which feels easiest to use from user perspective but is not unreasonably slow - I think for small matrix like this it doesn't matter if you draw by pixel or screen but for larger one by pixel might be a bit too slow but if library is not meant for large screens then that argument goes away - I think reasonably largest screen would be 64x64 so that gives you about 4k calls per 20ms which is around 200k operations per second - that might be good enough for Raspberry Pi 3. If we pick DrawPixel now worst case scenario we can always add extra overload to the scroller in the future with entire screen callback and just translate DrawPixel into DrawBuffer callback (basically you'd wrap user passed callback with your own which utilizes that to draw a screen). If we do reverse it will be much harder to simplify this later

Reply:
The "text scroller" in this instance is very simple: It increments a value (in the timer events) that is the horizontal offset position of the pre-rendered text (in `SenseHatTextRenderMatrix`) from where bits should be copied to the LED matrix display. It may be possible to generalize the use for small, single-line LED displays. Larger displays will probably lead to different design choices. My suggestion would be to possibly add an issue "generalize text rendering and scrolling for single line LED displays" or similar, that could be picked up by someone who has such a device, otherwise there may be a bit of guessing of what is feasible. This pull request could go into a different branch in the mean time, but I think it would not be too much work to refactor for generalization later. I'd be happy to test the generalized solution on SenseHat if/when available.

Reply:
There was a noticeable difference (smoother scrolling) when setting the pixels with a single `Write` operation vs multiple `SetPixel` (Pi 4, 8GB), although I can't recall whether I have visually compared in Debug mode only.  Maybe optimization here could also be left until implementation for another LED display is attempted?